move install functionality from custom to kubectl apply

JazminGonzalez-Rivero opened the pull request:
First pass at #254

this:
1) removes unnecessary manifests 
2) removes install / uninstall files and directory (moves manifest directory up a layer)
3) sets password to "password" at server startup if not previously set
4) warns users if they are still using the default password 
5) updated getting started docs

next steps:
default password should be more secure, (based on pod)
create API endpoint for updating password

alexmt commented on test/e2e/fixture.go:
```diff
@@ -90,12 +89,10 @@ func getFreePort() (int, error) {
 
 func (f *Fixture) setup() error {
-	installer, err := install.NewInstaller(f.Config, install.InstallOptions{})
+	_, err := exec.Command("kubectl apply -f ../../manifests/.").Output()
```
e2e tests require only CRDs, config maps/secrets

JazminGonzalez-Rivero replied:
@alexmt cool, was wondering if It was on to explicitly call out 

01a_application-crd.yaml and 01b_appproject-crd.yaml since these files names can change, but i guess I can keep them in a var so we can easily change them as needed so we are not loading in all the manifests.

JazminGonzalez-Rivero commented:
spoke with Jesse during "standup" some feedback:

1)for the manifests we should drop the namespace object so it is not made under default each time, 
- to do this we will need to update the "getting started" to say ` kubectl apply -f . --namespace argocd `

2)Re-add in the 2 dummy manifests I removed (02b). this allows the secret logic in the API server to differentiate between an empty secret and a missing secret. 
- if a secret is missing we should throw a fatel saying "secret is missing" 
- if secret is present but empty, auto generate as now.

3) also re-add (02a) we want to have dummy values for cmd. 
- handle as needed

JazminGonzalez-Rivero commented:
@jessesuen @merenbach can someone please review? 
since last review i...
re-added 02a and 02b 
1) server boot up fails if secrets are missing 
2) server boot-up sets up default values if secrets are blank
3) getting started readme now says to pass the namespace ``` kubectl apply -f . --namespace argocd ```
4) 02a now references the sso docs and sso docs are now more comprehensive 
5) e2e test now only loads in needed manifests rather than all manifests

merenbach commented on server/server.go:
```diff
@@ -103,10 +105,60 @@ type ArgoCDServerOpts struct {
+
+	if cdSettings.LocalUsers[common.ArgoCDAdminUsername] == "" {
+		//placeholder for when we replace this with get pod
+		passwordRaw := "password"
```
@JazminGonzalez-Rivero @jessesuen I'm not entirely sure I understand this part.  I see the bit about it being a placeholder, but not sure what the ultimate aim is.  Might it be more secure to create the map below (line 128) with a hash of `"*"` (i.e., an invalid, unreachable hash; my understanding is that an asterisk is somewhat common for this) so that no login is possible until/unless the password becomes set?

jessesuen replied:
1. I think this logic might need to change following my previous suggestion to start out with empty secret. I believe settingsMgr.GetSettings() will only work when there is a fully populated settings.

2. lets name this function initializeSettings().

3. I actually just realized that we can use hostname as the default password. In kubernetes, this translates to the pod name. This way we can start out with sane defaults. 

That said, If the pod restarts before they have a chance to change the password, the new pod name will not reflect the password.

merenbach commented on server/server.go:
```diff
@@ -103,10 +105,60 @@ type ArgoCDServerOpts struct {
+		cdSettings.LocalUsers = map[string]string{
+			common.ArgoCDAdminUsername: hashedPassword,
+		}
+	} else if valid, _ := password.VerifyPassword("password", cdSettings.LocalUsers[common.ArgoCDAdminUsername]); valid {
```
Rather than checking for a default password, I'm wondering if just using an invalid hash (that can never be validated) might be more secure.  Maybe we could add an accessor to determine if the hash is equal to an asterisk (`== "*"`) to say if the account is disabled, then can just check if the default account is disabled.

jessesuen commented on docs/getting_started.md:
````diff
@@ -18,7 +18,8 @@ chmod +x /usr/local/bin/argocd
 ```
-argocd install
+cd manifests
+kubectl apply -f manifests/. --namespace argocd
````
As discussed, lets not change getting_started.md until we cut a release that people can follow. You can create a different issue/PR for updating getting_started.md once we have 0.6

jessesuen commented on manifests/02b_argocd-secret.yaml:
```diff
@@ -7,20 +7,20 @@ metadata:
 stringData:
   # bcrypt hash of the string "password"
-  admin.password: $2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W
+  admin.password: ""
```
Ever key in this file can be commented out. Basically, we should start out with an empty secret (vs. a secret with empty values).

During API server startup, after retrieving the `argocd-secret`, if the keys `admin.password` and `server.secretkey ` are *missing*, the API server should populate it. This is different than checking that it is an empty string.

The reason we want it to be missing vs empty string is because a subsequent `kubectl apply` of the yaml will clobber the values that the API server generated.

JazminGonzalez-Rivero commented:
ah @jessesuen  i miss-understood the requirements my bad.

 I changed it to now error when the configmap is missing but replace default on null. 
updated the classname 
and changed the default to be the hostname.
